src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -195,7 +195,7 @@ impl IntrinsicCallMethods<'tcx> for Builder<'a, 'll, 'tcx> {
                 // and/or monomorphization invalidates these assumptions.
                 let coverageinfo = tcx.coverageinfo(caller_instance.def_id());
                 let mangled_fn = tcx.symbol_name(caller_instance);
-                let (mangled_fn_name, _len_val) = self.const_str(mangled_fn.name);
+                let (mangled_fn_name, _len_val) = self.const_str(mangled_fn);
```

Comment:
Is this why you changed the `const_str` API?

IMO `const_str` shouldn't care about this usecase, and if caching is desired, we probably want to cache the result of `const_cstr` (`const_str` is used here but the length is ignored anyway) once per `FunctionCx`, since the symbol name is that of the caller.

Reply:
I don't understand this comment. Are you asking for a specific change?

Reply:
I just pushed a new commit that reverts this part of the change, because it's not the important part. We end up with one extra `Symbol::intern()` call as a result.